payjoin/src/uri/error.rs

```diff
@@ -52,3 +55,41 @@ impl std::fmt::Display for PjParseError {
         }
     }
 }
+
+impl Error for PjParseError {}
+
+#[derive(Debug)]
+pub struct PayjoinUriError {
+    message: String,
+    source: Option<Box<dyn Error + Send + Sync>>,
+}
+
+impl PayjoinUriError {
+    pub fn new<S: Into<String>>(message: S) -> Self {
+        Self { message: message.into(), source: None }
+    }
+
+    pub fn from_uri_error(error: bitcoin_uri::de::Error<PjParseError>) -> Self {
+        Self { message: format!("Bitcoin URI error: {}", error), source: None }
+    }
```

Comment:
You can use variables right inside the format! braces . not doing so will most likely  break the lint test on CI